--- distro-theme-1.4.40/Makefile.0000~	2016-07-25 22:29:06.000000000 +0200
+++ distro-theme-1.4.40/Makefile	2016-09-23 19:33:51.121171255 +0200
@@ -40,6 +40,7 @@ install:
 	  install -m644 $$t/plymouth/*.script $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/; \
 	  install -m644 $$t/plymouth/*.plymouth $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/; \
 	  install -m644 $$t/plymouth/*.png $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/; \
+	  if [ "$(NO_GRUB)" != "true" ]; then \
 	  install -d $(DESTDIR)$(prefix)$(sharedir)/gfxboot/themes/$$t;  \
 	  install -d $(DESTDIR)/boot/grub2/themes/$$t; \
 	  install -d $(DESTDIR)/boot/grub2/themes/$$t/icons; \
@@ -55,23 +56,24 @@ install:
 	  if [ -d $$t/icons/gfxboot/ ]; then \
 	  install -m644 $$t/icons/gfxboot/*.* $(DESTDIR)/boot/grub2/themes/$$t/icons; \
 	  fi; \
+	  fi; \
 	  if [ ! -e $$t/gfxboot/background.png ] &&  [ ! -e $$t/plymouth/background.png ] && [ ! -e $$t/plymouth/suspend.png ]; then \
 	    if [ -e $$t/background/$$t-$(DEFAULT_RES).png ]; then \
-		    install -m644 $$t/background/$$t-$(DEFAULT_RES).png $(DESTDIR)/boot/grub2/themes/$$t/background.png; \
+		    [ "$(NO_GRUB)" != "true" ] && install -m644 $$t/background/$$t-$(DEFAULT_RES).png $(DESTDIR)/boot/grub2/themes/$$t/background.png; \
 		    install -m644 $$t/background/$$t-$(DEFAULT_RES).png $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/background.png; \
 		    convert -colorspace Gray $$t/background/$$t-$(DEFAULT_RES).png $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/suspend.png; \
 	    elif [ -e $$t/background/$$t-$(FALLBACK_RES).png ]; then \
-		    install -m644 $$t/background/$$t-$(FALLBACK_RES).png $(DESTDIR)/boot/grub2/themes/$$t/background.png; \
+		    [ "$(NO_GRUB)" != "true" ] && install -m644 $$t/background/$$t-$(FALLBACK_RES).png $(DESTDIR)/boot/grub2/themes/$$t/background.png; \
 		    install -m644 $$t/background/$$t-$(FALLBACK_RES).png $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/background.png; \
 		    convert -colorspace Gray $$t/background/$$t-$(FALLBACK_RES).png $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/suspend.png; \
 	    fi; \
 	    if [ -e $$t/background/$$t-$(DEFAULT_RES).jpg ]; then \
-		    convert $$t/background/$$t-$(DEFAULT_RES).jpg $(DESTDIR)/boot/grub2/themes/$$t/background.png \
-		    && convert $$t/background/$$t-$(DEFAULT_RES).jpg $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/background.png; \
+		    [ "$(NO_GRUB)" != "true" ] && convert $$t/background/$$t-$(DEFAULT_RES).jpg $(DESTDIR)/boot/grub2/themes/$$t/background.png \
+		    ; convert $$t/background/$$t-$(DEFAULT_RES).jpg $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/background.png; \
 		    convert -colorspace Gray $$t/background/$$t-$(DEFAULT_RES).jpg $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/suspend.png; \
 	    elif [ -e $$t/background/$$t-$(FALLBACK_RES).jpg ]; then \
-		    convert $$t/background/$$t-$(FALLBACK_RES).jpg $(DESTDIR)/boot/grub2/themes/$$t/background.png \
-		    && convert $$t/background/$$t-$(FALLBACK_RES).jpg $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/background.png; \
+		    [ "$(NO_GRUB)" != "true" ] && convert $$t/background/$$t-$(FALLBACK_RES).jpg $(DESTDIR)/boot/grub2/themes/$$t/background.png \
+		    ; convert $$t/background/$$t-$(FALLBACK_RES).jpg $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/background.png; \
 		    convert -colorspace Gray $$t/background/$$t-$(FALLBACK_RES).jpg $(DESTDIR)$(prefix)$(sharedir)/plymouth/themes/$$t/suspend.png; \
 	    fi; \
 	  fi; \
--- distro-theme-1.4.40/OpenMandriva/Makefile.0000~	2016-07-25 22:29:06.000000000 +0200
+++ distro-theme-1.4.40/OpenMandriva/Makefile	2016-09-23 19:40:17.777445421 +0200
@@ -1,8 +1,6 @@
-SUBDIRS=gfxboot
-
-.PHONY: subdirs $(SUBDIRS) clean
+ifeq ($(NO_GRUB),true)
+subdirs:
 
-subdirs: $(SUBDIRS)
-
-$(SUBDIRS):
-	@exec $(MAKE) -C $@ $(MAKECMDGOALS)
+else
+SUBDIRS=gfxboot
+endif
